### ISSUE TITLE
API documentation regarding filter creation is unclear

### ISSUE THREAD

**Issue by bocops** [mastodon/documentation#1216]
[Create a filter](https://docs.joinmastodon.org/methods/filters/#create) states that an integer value for `expires_in` should be given, and that this value will be interpreted as "How many seconds from now should the filter expire?". Documentation neither states that this value is optional, nor that some specific value (I assume "0") will be interpreted as a request for a filter that does not expire.

Non-expiring filters do exist, the [Filter](https://docs.joinmastodon.org/entities/Filter/) entity has an `expires_at` String value that is "null if the filter does not expire".

Please add documentation detailing how to create a non-expiring filter.

**Comment by trwnh** [comment]
anything not marked "required" is optional

**Comment by bocops** [comment]
> anything not marked "required" is optional

Makes sense given the "REQUIRED" tag on some values. I was probably confused switching between methods and entities while implementing some of it, because the latter explicitly marks some of its values as "OPTIONAL". I couldn't find this spelled out on either [Getting started with the API](https://docs.joinmastodon.org/client/intro/) and [Guidelines and best practices](https://docs.joinmastodon.org/api/guidelines/), so unless I've searched in the wrong locations, I'd suggest making this explicit at least once, somewhere.

> i suppose default values could be called out, though?

Definitely. I'm working on a Kotlin/Java Mastodon library, and trying to find all the implicit default values in the server code is quite hard at times.

**Comment by trwnh** [comment]
the following callouts are used throughout the api docs:

### methods
- "required" = if you don't provide this for a method, the method call will fail

### entities
- "optional" = the entity attr may not be present
- "nullable" = the entity attr may be present as an explicit null value

---

i can probably add this clarification to the pages you mentioned above, yeah

**Comment by bocops** [comment]
That would be nice, especially the idea of an onHover tooltip sounds great!

**Comment by andypiper** [comment]
@trwnh I'm going through to label API-related docs issues; I'm very happy if you want to pick this up ahead of me, however - you have much more familiarity with the docs and API at this time 👍🏻
